$OpenBSD: patch-bytes_cpp,v 1.1.1.1 2009/04/17 17:56:50 sthen Exp $

- unbreak all big-endian arch (first hunk)
- unbreak strict alignment big-endian (second hunk)

--- bytes.cpp.orig	Wed Mar 11 21:33:48 2009
+++ bytes.cpp	Fri Apr 17 18:43:52 2009
@@ -30,10 +30,10 @@ void WriteNumber(char *data, double dVal)
 	uint64_t res;
 
 #if __FLOAT_WORD_ORDER == __BYTE_ORDER
+        uint64_t in  = *((uint64_t*)&dVal);
 #if __BYTE_ORDER == __BIG_ENDIAN
-	res = dVal;
+	res = in;
 #elif __BYTE_ORDER == __LITTLE_ENDIAN
-        uint64_t in  = *((uint64_t*)&dVal);
         res = __bswap_64(in);
 #endif
 #else
@@ -60,7 +60,9 @@ double ReadNumber(const char *data)
 {
 #if __FLOAT_WORD_ORDER == __BYTE_ORDER
 #if __BYTE_ORDER == __BIG_ENDIAN
-	return *((double *)data);
+	double val;
+	memcpy(&val, data, sizeof(double));
+	return val;
 #elif __BYTE_ORDER == __LITTLE_ENDIAN
 	uint64_t in  = *((uint64_t*)data);
 	uint64_t res = __bswap_64(in);
